app/src/main/java/com/sjh/autosummary/feature/main/MainViewModel.kt

```diff
@@ -1,13 +1,214 @@
 package com.sjh.autosummary.feature.main
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sjh.autosummary.core.common.LoadState
+import com.sjh.autosummary.core.data.model.ChatRequest
 import com.sjh.autosummary.core.data.repository.ChatRepository
 import com.sjh.autosummary.core.data.repository.HistoryRepository
+import com.sjh.autosummary.core.domain.UpdateChatSummaryUseCase
+import com.sjh.autosummary.core.model.ChatHistory
+import com.sjh.autosummary.core.model.ChatRoleType
+import com.sjh.autosummary.core.model.MessageContent
+import com.sjh.autosummary.feature.main.contract.event.MainScreenEvent
+import com.sjh.autosummary.feature.main.contract.sideeffect.MainScreenSideEffect
+import com.sjh.autosummary.feature.main.contract.state.MainScreenState
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import org.orbitmvi.orbit.Container
+import org.orbitmvi.orbit.ContainerHost
+import org.orbitmvi.orbit.annotation.OrbitInternal
+import org.orbitmvi.orbit.viewmodel.container
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
 import javax.inject.Inject
 
+@OptIn(OrbitInternal::class)
 @HiltViewModel
 class MainViewModel @Inject constructor(
     private val chatRepository: ChatRepository,
     private val historyRepository: HistoryRepository,
-) : ViewModel()
+    private val updateChatSummaryUseCase: UpdateChatSummaryUseCase,
+) : ViewModel(),
+    ContainerHost<MainScreenState, MainScreenSideEffect> {
+
+    override val container: Container<MainScreenState, MainScreenSideEffect> =
+        container(initialState = MainScreenState())
+
+    fun handleEvent(event: MainScreenEvent) {
+        when (event) {
+            is MainScreenEvent.StartChat -> {
+                if (event.chatHistoryId != 0L) {
+                    loadChatHistory(event.chatHistoryId)
+                } else {
+                    createChatHistory()
+                }
+            }
+
+            is MainScreenEvent.OnSearchClick -> {
+                searchContent(event.content)
+            }
+
+            MainScreenEvent.OnHistoryClick -> {
+                saveChatHistory()
+            }
+        }
+    }
+
+    private fun saveChatHistory() {
+        viewModelScope.launch {
+            container.orbit {
```

Comment:
```suggestion
    private fun saveChatHistory(): Unit = viewModelScope.launch {
        container.orbit {
```
이렇게 하면 함수 indentation depth가 한 단계 줄어들 것 같은데 되나요?

Reply:
아 맞습니다. 뭔가 Unit을 다르게 생각해왔던 것 같습니다

Reply:
말씀해주신대로 작성해보니 launch 의 반환값이 잡혀 Job을 반환 타입으로 설정하면 depth를 줄일 수 있었습니다!